### PR TITLE
Feature/sr 20 firebase integration

### PR DESCRIPTION
**Pull request by GalacticOctopus**
- Add Firebase Auth and Firestore
- Create queries
- Attach to featured page

### CODE REVIEW

**Review by mzajaczkowsky — COMMENTED** [review]
Pull request adds Firebase to the project and updates the code of slider on Feature component.

File: src/hooks/useAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  return {
+    setEmail,
+    setPassword,
+    register,
+    logout,
+    login,
+  };
+};
```
Comment (masterHAWK99) [inline src/hooks/useAuth.tsx:44]: Please log potential errors by console.log, users don't have to see them everytime. What about checking if user is logged in?

File: src/hooks/useAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+{
+  /* Use case
+<input placeholder="Email" onChange={(e) => setEmail(e.target.value)} />
+<input placeholder="Password" type="password" onChange={(e) => setPassword(e.target.value)}></input>
+<button onClick={login}>Sign In</button>
+<button onClick={register}>Register</button>
+<button onClick={logout}>Sign Out</button> 
+*/
```
Comment (masterHAWK99) [inline src/hooks/useAuth.tsx:53]: Please don't leave commented code

File: src/hooks/useProducts.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  getDocs,
+  collection,
+  limit,
+  orderBy,
+  Query,
+  CollectionReference,
+} from "firebase/firestore";
+
+export const useProducts = () => {
+  const [productList, setProductList] = useState<any>([]);
```
Comment (masterHAWK99) [inline src/hooks/useProducts.tsx:15]: Add correct type instead of `any`

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

**Review by masterHAWK99 — APPROVED** [review]

**Pull request merged**